examples/tensorflow/segmentation/evaluation.py

```diff
@@ -164,41 +177,56 @@ def test_step(dataset_inputs):
-    """Runs evaluation on checkpoint save directory"""
-    strategy = get_distribution_strategy(config)
-    if config.metrics_dump is not None:
-        write_metrics(0, config.metrics_dump)
-
-    dataset_builder = get_dataset_builders(config, strategy.num_replicas_in_sync)
-    dataset = dataset_builder.build()
-    num_batches = dataset_builder.steps_per_epoch
-    test_dist_dataset = strategy.experimental_distribute_dataset(dataset)
-
-    # We use `model_batch_size` to create input layer for model
-    config.model_batch_size = dataset_builder.batch_size
-
-    model_builder = get_model_builder(config)
+def restore_compressed_model(config, strategy, model_builder, ckpt_path = None):
+    compression_state = None
+    if ckpt_path:
+        compression_state = load_compression_state(ckpt_path)
 
     with TFOriginalModelManager(model_builder.build_model,
                                 weights=config.get('weights', None),
                                 is_training=False) as model:
         with strategy.scope():
             compression_ctrl, compress_model = create_compressed_model(model,
                                                                        config.nncf_config,
-                                                                       should_init=False)
+                                                                       compression_state)
+
             variables = get_variables(compress_model)
             checkpoint = tf.train.Checkpoint(variables=variables,
-                                             compression_ctrl=compression_ctrl,
+                                             compression_state=TFCompressionState(compression_ctrl),
                                              step=tf.Variable(0))
-            eval_metric = model_builder.eval_metrics()
-            predict_post_process_fn = model_builder.post_processing
+            if ckpt_path:
+                load_checkpoint(checkpoint, config.ckpt_path)
 
-    test_step = create_test_step_fn(strategy, compress_model, predict_post_process_fn)
+    return compression_ctrl, compress_model, checkpoint
+
+
+def run_evaluation(config, eval_timeout=None):
+    """Runs evaluation on checkpoint save directory"""
+    strategy = get_distribution_strategy(config)
+    if config.metrics_dump is not None:
+        write_metrics(0, config.metrics_dump)
+
+    validation_builder, calibration_builder = get_dataset_builders(config, strategy.num_replicas_in_sync)
+    calibration_dataset = calibration_builder.build()
+    val_dataset = validation_builder.build()
+    num_batches = validation_builder.steps_per_epoch
+    test_dist_dataset = strategy.experimental_distribute_dataset(val_dataset)
+
+    config.nncf_config = register_default_init_args(nncf_config=config.nncf_config,
+                                                    data_loader=calibration_dataset,
+                                                    batch_size=validation_builder.global_batch_size)
```

Comment:
Registered init args to pass sanity tests on mask-rcnn when evaluation is run with bn adapt and range init

Previously, init was forcely disabled by should_init=False and no checkpoint was passed, which is impossible to do now without hacks.